File: ProyectoCivitasJava/src/civitas/CasillaCalle.java
/**
 * Proyecto Civitas
 * Sergio Hervás Cobo
 * Javier Suárez Lloréns
 * Programación y Diseño Orientado a Objetos
 * 2º B3 - Grado en Ingeniería Informática
 * E.T.S.I. Informática y de Telecomunicación
 * Universidad de Granada
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author sergio
 */
public class CasillaCalle extends Casilla {
    
    private TituloPropiedad tituloPropiedad;
    
    CasillaCalle(TituloPropiedad titulo){
        super(titulo.getNombre());
        this.tituloPropiedad = titulo;       
    }
   
    //visibilidad cambiada
    public TituloPropiedad getTituloPropiedad(){
        return tituloPropiedad;
    }    
    
    @Override
    void recibeJugador(int iactual, ArrayList <Jugador> todos){
       if(jugadorCorrecto(iactual, todos)){
           informe(iactual, todos);
           Jugador jugador = todos.get(iactual);
           
           if (!tituloPropiedad.tienePropietario()){
               jugador.puedeComprarCasilla();
           }
           else{
               tituloPropiedad.tramitarAlquiler(jugador);
           } 
       }
    }
    
    @Override
    public String toString(){
        String salida = getNombre() + " | " + tituloPropiedad;
        return salida;
    }
    
}

File: ProyectoCivitasRuby/lib/modelo/casilla_calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class CasillaCalle < Casilla
    
    def initialize(titulo)
      super(titulo.nombre)
      @titulopropiedad = titulo;
    end
    
    attr_reader :titulopropiedad;
    
    #@Override
    def recibe_jugador(iactual, todos)
      if(jugador_correcto(iactual,todos))
        informe(iactual, todos)
        jugador=todos.at(iactual)
        if(@titulopropiedad.tiene_propietario()==false)
          jugador.puede_comprar_casilla()
        else
          @titulopropiedad.tramitar_alquiler(jugador)
        end
      end
    end
    
    #@Override    
    def to_s
      salida = "#{@nombre} | #{@titulopropiedad}"
    end
  end
end